content/_code-samples/build-a-wallet/desktop-js/library/5_helpers.js

```diff
@@ -0,0 +1,139 @@
+        reserve = prepareReserve(rawLedgerData)
+        const ledger = prepareLedgerData(rawLedgerData)
+        appWindow.webContents.send('update-ledger-data', ledger)
+    })
+
+    // Wait for transaction on subscribed account and re-request account data
+    client.on("transaction", async (transaction) => {
+        // Reference: https://xrpl.org/account_info.html
+        const accountInfoRequest = {
+            "command": "account_info",
+            "account": wallet.address,
+            "ledger_index": transaction.ledger_index
+        }
+
+        const accountInfoResponse = await client.request(accountInfoRequest)
+        const accountData = prepareAccountData(accountInfoResponse.result.account_data)
+        appWindow.webContents.send('update-account-data', accountData)
+
+        const transactions = prepareTxData([{tx: transaction.transaction}])
+        appWindow.webContents.send('update-transaction-data', transactions)
+    })
+}
+
+/**
+ * Saves the wallet seed using proper cryptographic functions
+ *
+ * @param WALLET_DIR
+ * @param seed
+ * @param password
+ */
+const saveSaltedSeed = (WALLET_DIR, seed, password)=> {
+    const salt = crypto.randomBytes(20).toString('hex')
+
+    fs.writeFileSync(path.join(__dirname, WALLET_DIR, 'salt.txt'), salt);
+
+    // Hashing salted password using Password-Based Key Derivation Function 2
+    const derivedKey = crypto.pbkdf2Sync(password, salt, 1000, 32, 'sha256')
+
+    // Generate a Fernet secret we can use for symmetric encryption
+    const secret = new fernet.Secret(derivedKey.toString('base64'));
+
+    // Generate encryption token with secret, time and initialization vector
+    // In a real-world use case we would have current time and a random IV,
+    // but for demo purposes being deterministic is just fine
+    const token = new fernet.Token({
+        secret: secret,
+        time: Date.parse(1),
+        iv: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
+    })
+
+    const privateKey = token.encode(seed)
+
+    fs.writeFileSync(path.join(__dirname, WALLET_DIR, 'seed.txt'), privateKey)
+}
+
+/**
+ * Loads the plaintext value of the encrypted seed
+ *
+ * @param WALLET_DIR
+ * @param password
+ * @returns {*}
+ */
+const loadSaltedSeed = (WALLET_DIR, password) => {
+    const salt = fs.readFileSync(path.join(__dirname, WALLET_DIR, 'salt.txt')).toString()
+
+    const encodedSeed = fs.readFileSync(path.join(__dirname, WALLET_DIR, 'seed.txt')).toString()
+
+    // Hashing salted password using Password-Based Key Derivation Function 2
+    const derivedKey = crypto.pbkdf2Sync(password, salt, 1000, 32, 'sha256')
+
+    // Generate a Fernet secret we can use for symmetric encryption
+    const secret = new fernet.Secret(derivedKey.toString('base64'));
+
+    // Generate decryption token
+    const token = new fernet.Token({
+        secret: secret,
+        token: encodedSeed,
+        ttl: 0
+    })
+
+    return token.decode();
```

Comment:
I'd recommend adding a basic error catch here if the user inputs the wrong password, because currently it throws an uncaught exception when trying to decode with it, and then the user has to stop and start the app again to input a new password.

Reply:
Bump on this